chainstate/tx-verifier/src/transaction_verifier/timelock_check.rs

```diff
@@ -167,88 +159,9 @@ where
         Ok(())
     })?;
 
-    // check if output timelocks follow the rules
-    if let Some(outputs) = tx.outputs() {
-        // in case `CreateStakePool`, `ProduceBlockFromStake` utxos are spent or an input from account
-        // then produced outputs must be timelocked as per chain config
-        let output_check_required = inputs
-            .iter()
-            .zip(input_utxos.iter())
-            .filter_map(|(input, utxo_with_outpoint)| match input {
-                TxInput::Utxo(_) => {
-                    match utxo_with_outpoint.clone().expect("must be present").1.output() {
-                        TxOutput::Transfer(_, _)
-                        | TxOutput::LockThenTransfer(_, _, _)
-                        | TxOutput::Burn(_)
-                        | TxOutput::CreateDelegationId(_, _)
-                        | TxOutput::DelegateStaking(_, _) => None,
-                        TxOutput::CreateStakePool(_, _) | TxOutput::ProduceBlockFromStake(_, _) => {
-                            Some(OutputTimelockCheckRequired::DecommissioningMaturity)
-                        }
-                    }
-                }
-                TxInput::Account(account_input) => match account_input.account() {
-                    AccountSpending::Delegation(_, _) => {
-                        Some(OutputTimelockCheckRequired::DelegationSpendMaturity)
-                    }
-                },
-            })
-            // Note: theoretically there could be several inputs that impose timelock requirement
-            // but for now such transaction is considered invalid
-            .at_most_one()
-            .map_err(|_| ConnectTransactionError::InvalidInputTypeInTx)?;
-
-        if let Some(output_check_required) = output_check_required {
-            check_outputs_timelock(
-                chain_config.as_ref(),
-                outputs,
-                tx_source.expected_block_height(),
-                output_check_required,
-                outpoint_source_id,
-            )?;
-        }
-    }
-
     Ok(())
 }
 
-/// Outputs that decommission a stake pool or spend delegation share must be timelocked
-fn check_outputs_timelock(
```

Comment:
I'm keeping this comment as a reference to see why this was removed

Reply:
This check was introduced to validate that if input decommissions the pool or spends share then all outputs must be locked according to the chain config. ConstraintsAccumulator has the same functionality making this function obsolete.